src/java.base/share/classes/jdk/internal/vm/annotation/constant-folding.md

```diff
@@ -0,0 +1,128 @@
+Constant Folding in the Hotspot Compiler
```

Comment:
I assume any write-up of HotSpot constant folding should move into src/hotspot tree, maybe a block comment in one of the source files?